hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/metrics/TestSystemMetricsPublisher.java

```diff
@@ -72,9 +72,21 @@
 
 public class TestSystemMetricsPublisher {
 
-  protected static ApplicationHistoryServer timelineServer;
-  protected static TimelineServiceV1Publisher metricsPublisher;
-  protected static TimelineStore store;
+  public static void setTimelineServer(ApplicationHistoryServer timelineServer) {
+    TestSystemMetricsPublisher.timelineServer = timelineServer;
+  }
+
+  public static void setMetricsPublisher(TimelineServiceV1Publisher metricsPublisher) {
+    TestSystemMetricsPublisher.metricsPublisher = metricsPublisher;
+  }
+
+  public static void setStore(TimelineStore store) {
+    TestSystemMetricsPublisher.store = store;
+  }
```

Comment:
The methods can be package-private (i.e. public should be removed).

Reply:
yes